### PR TITLE
Added Center widget

### PR DESCRIPTION
**Pull request by vinothvino42**
## Description

In this PR, I added a center mirai parser and added an example route for the center widget in the mirai gallery app.

## Related Issues

NONE

## Type of Change
- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: packages/mirai/lib/src/utils/icon_utils.dart [hunk 1]
```diff
@@ -1923,6 +1923,7 @@ final Map<String, IconData> materialIconMap = {
   'album_sharp': Icons.album_sharp,
   'album_rounded': Icons.album_rounded,
   'album_outlined': Icons.album_outlined,
+  'format_align_center': Icons.format_align_center,
```
Comment (divyanshub024) [inline packages/mirai/lib/src/utils/icon_utils.dart:1926]: I think this icon is already there.

**Review by divyanshub024 — CHANGES_REQUESTED** [review]

File: packages/mirai/lib/src/utils/icon_utils.dart (hunk 1 above)
Comment (vinothvino42) [reply packages/mirai/lib/src/utils/icon_utils.dart]: @divyanshub024 Ohh sorry, I removed it. Can you check it again?

File: examples/mirai_gallery/assets/json/center_example.json [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                "color": "#800000",
+                "child": {
+                    "type": "text",
+                    "data": "Flutter",
+                    "style": {
+                        "color": "#ffffff",
+                        "fontSize": 23
+                    }
+                },
+                "onPressed": {}
```
Comment (divyanshub024) [inline examples/mirai_gallery/assets/json/center_example.json:31]: Can we remove the `onPressed` here? As the container does not take onpressed.

File: examples/mirai_gallery/assets/json/center_example.json [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    "type": "scaffold",
+    "appBar": {
+        "type": "appBar",
+        "title": {
+            "type": "text",
+            "data": "Center"
+        }
+    },
+    "body": {
+        "type": "container",
```
Comment (divyanshub024) [inline examples/mirai_gallery/assets/json/center_example.json:11]: Can we remove this container so child is always in the center of the screen?

**Review by divyanshub024 — CHANGES_REQUESTED** [review]
@vinothvino42 Thanks for amazing work 💯 

Please check my comments.

File: examples/mirai_gallery/assets/json/center_example.json (hunk 2 above)
Comment (vinothvino42) [reply examples/mirai_gallery/assets/json/center_example.json:31]: My bad, I just copied and pasted from another example. I'll fix it now.

File: examples/mirai_gallery/assets/json/center_example.json (hunk 3 above)
Comment (vinothvino42) [reply examples/mirai_gallery/assets/json/center_example.json:11]: I added it for showing how the center widget's `widthFactor` and `heightFactor` property works. Okay, I'll remove it and push the code now.

File: examples/mirai_gallery/assets/json/center_example.json (hunk 2 above)
Comment (divyanshub024) [reply examples/mirai_gallery/assets/json/center_example.json:31]: Thank you!!

File: examples/mirai_gallery/assets/json/center_example.json (hunk 3 above)
Comment (divyanshub024) [reply examples/mirai_gallery/assets/json/center_example.json:11]: Hmm, it's better to remove it. Bcz people might run it in a web/desktop app and while resizing it's good to see the widget resize as well.

Comment (vinothvino42) [reply examples/mirai_gallery/assets/json/center_example.json:11]: Okay, I'll remove the top Container widget and widthFactor and heightFactor from the Center widget.

**Review by DivyanshuBhargavaSecurrency — APPROVED** [review]

**Pull request merged**